src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -172,89 +220,169 @@ class instance_scarlet_monastery : public InstanceMapScript
                     return WhitemaneGUID;
                 case DATA_DOOR_WHITEMANE:
                     return DoorHighInquisitorGUID;
+                case DATA_DOOR_CHAPEL:
+                    return DoorChapelGUID;
+                default:
+                    return ObjectGuid::Empty;
+                    break;
             }
-
-            return ObjectGuid::Empty;
         }
 
         uint32 GetData(uint32 type) const override
         {
-            if (type == TYPE_MOGRAINE_AND_WHITE_EVENT)
-                return encounter;
-            else if (type == DATA_HORSEMAN_EVENT)
-                return encounter;
-            return 0;
+            switch (type)
+            {
+                case TYPE_MOGRAINE_AND_WHITE_EVENT:
+                    return encounter;
+                    break;
+                case DATA_HORSEMAN_EVENT:
+                    return encounter;
+                    break;
+                case TYPE_ASHBRINGER_EVENT:
+                    return ashencounter;
+                    break;
+                default:
+                    return 0;
+                    break;
+            }
         }
     private:
         ObjectGuid DoorHighInquisitorGUID;
+        ObjectGuid DoorChapelGUID;
         ObjectGuid MograineGUID;
         ObjectGuid WhitemaneGUID;
-        uint32 encounter;
+        uint32 encounter{};
+        uint32 ashencounter{};
+        GuidSet AshbringerNpcGUID;
```

Comment:
Private members should use this style _privateMember

so _whitemaneGUID, _doorChapelGUID etc

Reply:
https://www.azerothcore.org/wiki/cpp-code-standards#naming-of-variables-and-functions